### PR TITLE
feature: open webpage in new window

### PR DESCRIPTION
**Pull request by ericawright**
open for visibility

tested with mac - `cmd-click` opens page in new tab.
Only works when running on electron.

TODO: 
- ~~ensure the tab opens beside the parent tab.~~
- should it open in the background or foreground? ( I need to be able to test with foreground opening for the desired animation, my understanding is that this can generally be user-set)

### CODE REVIEW

File: src/browser-frontend/sagas/webcontents-action-handlers.js [hunk 1]
```diff
@@ -94,8 +94,8 @@ function* onPageDidNavigateInternal() {
   // TODO
 }
 
-function* onPageDidNavigateToNewWindow() {
-  // TODO
+function* onPageDidNavigateToNewWindow({ payload: { pageId, url } }) {
+  yield put(PagesModelActions.addPage({ url, pageId }));
```
Comment (victorporof) [inline src/browser-frontend/sagas/webcontents-action-handlers.js]: Nit: it's useless to pass in the pageId here to the addPage action, which doesn't use it. That id is the id of the page where the navigation occured, and every single page/tab needs to have a unique id. The reducer handling this addPage action will create a new id for the newly added page/tab. Currently we're not tracking tab parents, but if we did remembering this old id here would be a good idea.

**Review by victorporof — COMMENTED** [review]
This is great! I'm ok with even landing this as is, but we'll need to handle the "disposition" flag to open in foreground/background or a new window etc.

File: src/browser-frontend/sagas/webcontents-action-handlers.js (hunk 1 above)
Comment (ericawright) [reply src/browser-frontend/sagas/webcontents-action-handlers.js]: I'm still passing it, but now as `parentId` to position the tab beside the parent.

**Comment by ericawright** [comment]
is it okay with you if I default it to always be a foreground tab?

**Comment by victorporof** [comment]
Fine for now! Landing.

File: src/browser-frontend/reducers/pages-reducers.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ function addPage(state, { payload: { url, index, background } = {} }) {
     mut.updateIn(['ui', 'pages', 'visuals'], m => m.set(pageId, pageUIState));
 
     const pageCount = state.ui.pages.displayOrder.count();
-    const pageIndex = index != null ? index : pageCount;
+    const pageIndex = parentId ? state.ui.pages.displayOrder.findIndex(id => id === parentId) + 1 : pageCount;
```
Comment (victorporof) [inline src/browser-frontend/reducers/pages-reducers.js:35]: Nit: This doesn't handle the case where `parentId` isn't in `displayOrder`. I bet that can never happen now, but never trust your caller etc.

**Pull request merged**